### PR TITLE
feat: add tips and download button

### LINKED ISSUE #13: Add help for figuring out the version you're on
We should add a help button/tooltip somewhere to help users figure out what version they're on.
The version is in package.json in `ct3aMetadata`

### LINKED ISSUE #18: Add download button for diffs
On [slug].tsx page, we should add a download button next to each diff so it's easy for people to apply the diff. Then maybe a tip on how to apply patches.

### PR DESCRIPTION
**Pull request by PiotrekPKP**
### Changelog
- **adds tip to let users know how to check current app version** (closes #13)
- adds dialog and button components from shadcn and adds additional config for them to work ([there has been an update to the installation docs](https://ui.shadcn.com/docs/installation#configure-tailwindconfigjs), new things in the `tailwind.config.ts` were required)
- **adds download button and `.patch` application tip** (closes #18)
- adds loading state to the upgrade button (it looked laggy when the diff page was being generated for longer time)

### CODE REVIEW

File: src/components/ui/Dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+DialogPortal.displayName = DialogPrimitive.Portal.displayName;
+
+const DialogOverlay = React.forwardRef<
+  React.ElementRef<typeof DialogPrimitive.Overlay>,
+  React.ComponentPropsWithoutRef<typeof DialogPrimitive.Overlay>
+>(({ className, ...props }, ref) => (
+  <DialogPrimitive.Overlay
+    ref={ref}
+    className={cn(
+      "bg-background/80 fixed inset-0 z-50 backdrop-blur-sm transition-all duration-100 data-[state=closed]:animate-out data-[state=closed]:fade-out data-[state=open]:fade-in",
```
Comment (andreifilip123) [inline src/components/ui/Dialog.tsx]: I think the background for the dialog is a bit too opaque. Could we lower the opacity for it ? (I'm guessing it's the bg-background/80 class)

**Review by andreifilip123 — APPROVED** [review]
Looks really good. Thank you for this 🙏 Can you change the overlay color a bit ? Apart from that, it's good

File: src/components/ui/Dialog.tsx (hunk 1 above)
Comment (PiotrekPKP) [reply src/components/ui/Dialog.tsx]: Is 60 ok?

image

Comment (andreifilip123) [reply src/components/ui/Dialog.tsx]: Try even lower 😅

Comment (PiotrekPKP) [reply src/components/ui/Dialog.tsx]: Can also go for 30 :)

image

Comment (andreifilip123) [reply src/components/ui/Dialog.tsx]: Let's go with 40 and merge this 👌

Comment (PiotrekPKP) [reply src/components/ui/Dialog.tsx]: Pushed :)

Prettier also reordered classes for whatever reason :shrug:

**Review by andreifilip123 — APPROVED** [review]

**Pull request merged**